BuyPolish/Pola/UI/ProductSearch/ScanCode/ScanCodeViewController.swift

```diff
@@ -196,3 +218,40 @@ extension ScanCodeViewController: ResultsViewControllerDelegate {
         castedView.setButtonsVisible(true, animated: true)
     }
 }
+
+extension ScanCodeViewController: UIImagePickerControllerDelegate {
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey: Any]) {
```

Comment:
Zastanawiam się czy dałoby radę wynieść całą tą część z wybieraniem zdjęcia i rozpoznawania do oddzielnego VC (np. `ScanImageFromGalleryViewController`)tak by `ScanCodeViewController` otwierał go tylko i w delegacie otrzymywał gotowy kod. Troche się rozrasta ten VC i dobrze by było dzielić kod na mniejsze VC.

Reply:
Taki pusty VC, który we `viewDidLoad` otwiera picker? hmmm zobaczmy

Reply:
Myślałem bardziej o dodaniu image pickera jako child do tego nowego. Nie wiem tylko czy będzie możliwe wepchniecie tam tego całego kodu bez żadnych dziwnych tricków.

Reply:
Czyli zrobienie pustego VC, bo nie będzie tam żadnego UI. Będą tylko delegaty i wywołanie pickera.

Reply:
hmmm no wymagałoby to chyba jakiś gimnastyk bo trzeba jeszcze zmieniać stan `galleryButton.isSelected` więc nie wiem czy nie byłoby to zejście z deszczu pod rynne...

Reply:
ew dodanie kilku delegat które i tak trzeba by zaimplementować w `ScanCodeViewController`